### PR TITLE
Never compile a regex for the empty string

### PR DESCRIPTION
**Pull request by t-wissmann**
An empty string is not allowed in the Posix grammar for extended regular
expressions. On openbsd, this led to a run time error on startup
because we initialize the keymask regexes with the empty string.

In the keymask code, we are using the empty string anyway as a special
case for an 'inactive' or 'unset' regular expression, so there is no
need to compile the regex for "".

Thanks to Lucas <lucas@sexy.is> and Florian Viehweger
<openbsd@out-of-creativity.de> for spotting and reporting this issue.

### CODE REVIEW

File: NEWS [hunk 1]
```diff
@@ -13,6 +13,9 @@ Current git version
   * keybind now checks that the bound command exists.
   * cycle_all (Alt-Tab) now also traverses floating clients
   * new setting 'auto_detect_panels' controlling the panel detection algorithm.
+  * Bug fixes:
+    - Handle the regular expression "" correctly, since the gramar extended
```
Comment (blarz) [inline NEWS:17]: grammar

**Comment by t-wissmann** [comment]
I don't know how to test it, except by adding a openbsd build to the CI.

**Comment by blarz** [comment]
> I don't know how to test it, except by adding a openbsd build to the CI.

But you could add a test testing that you get an error when trying to do something with an empty regex?

**Comment by t-wissmann** [comment]
> But you could add a test testing that you get an error when trying to do something with an empty regex?

I don't understand. The issue on openbox was that `std::regex("", …)` throws an exception. Obviously this does not happen in our CI build because otherwise *all* test cases would fail. So I don't see what I can test with our current CI setup. (Note that e.g. the regex `keymask` is `""` per default, so we've been checking the empty regex all the time)

**Comment by blarz** [comment]
> > But you could add a test testing that you get an error when trying to do something with an empty regex?
> 
> I don't understand. The issue on openbox was that `std::regex("", …)` throws an exception. Obviously this does not happen in our CI build because otherwise _all_ test cases would fail. So I don't see what I can test with our current CI setup. (Note that e.g. the regex `keymask` is `""` per default, so we've been checking the empty regex all the time)

Ah ok, sorry I got that wrong then, nevermind :)

**Comment by t-wissmann** [comment]
> Is it still wip?

I wanted to wait until tomorrow for the OpenBSD-guys to react. Apart from this, the PR is ready to be approved/merged IMO

**Comment by blarz** [comment]
> > Is it still wip?
> 
> I wanted to wait until tomorrow for the OpenBSD-guys to react. Apart from this, the PR is ready to be approved/merged IMO

Ok, let's wait then. Other than that this looks good to go.

**Comment by The-Compiler** [comment]
Looks like they're happy: [BUG: An empty regex is not allowed in the POSIX grammar.](https://lists.schokokeks.org/pipermail/hlwm/2020-April/000306.html)
